test/suites/integration/storage_test.go

```diff
@@ -16,104 +16,212 @@ package integration_test
 
 import (
 	"fmt"
+	"strings"
 
-	"github.com/aws/aws-sdk-go/aws"
 	appsv1 "k8s.io/api/apps/v1"
 	v1 "k8s.io/api/core/v1"
 	storagev1 "k8s.io/api/storage/v1"
-	"k8s.io/apimachinery/pkg/api/errors"
-	"k8s.io/apimachinery/pkg/api/resource"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	"sigs.k8s.io/controller-runtime/pkg/client"
 
+	"github.com/aws/aws-sdk-go/aws"
 	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	"github.com/samber/lo"
 
+	"github.com/aws/karpenter/pkg/errors"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
 	"sigs.k8s.io/karpenter/pkg/test"
 )
 
-// This test requires the EBS CSI driver to be installed
-var _ = Describe("Dynamic PVC", func() {
-	It("should run a pod with a dynamic persistent volume", func() {
-		// Ensure that the EBS driver is installed, or we can't run the test.
-		var ds appsv1.DaemonSet
-		if err := env.Client.Get(env.Context, client.ObjectKey{
-			Namespace: "kube-system",
-			Name:      "ebs-csi-node",
-		}, &ds); err != nil {
-			if errors.IsNotFound(err) {
-				Skip(fmt.Sprintf("skipping dynamic PVC test due to missing EBS driver %s", err))
-			} else {
-				Fail(fmt.Sprintf("determining EBS driver status, %s", err))
-			}
-		}
-		storageClassName := "ebs-sc-test"
-		bindMode := storagev1.VolumeBindingWaitForFirstConsumer
-		sc := test.StorageClass(test.StorageClassOptions{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: storageClassName,
-			},
-			Provisioner:       aws.String("ebs.csi.aws.com"),
-			VolumeBindingMode: &bindMode,
+var _ = Describe("Persistent Volumes", func() {
+	Context("Static", func() {
+		It("should run a pod with a pre-bound persistent volume (empty storage class)", func() {
+			pvc := test.PersistentVolumeClaim(test.PersistentVolumeClaimOptions{
+				VolumeName:       "test-volume",
+				StorageClassName: lo.ToPtr(""),
+			})
+			pv := test.PersistentVolume(test.PersistentVolumeOptions{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: pvc.Spec.VolumeName,
+				},
+			})
+			pod := test.Pod(test.PodOptions{
+				PersistentVolumeClaims: []string{pvc.Name},
+			})
+
+			env.ExpectCreated(nodeClass, nodePool, pv, pvc, pod)
+			env.EventuallyExpectHealthy(pod)
+			env.ExpectCreatedNodeCount("==", 1)
 		})
-
-		pvc := test.PersistentVolumeClaim(test.PersistentVolumeClaimOptions{
-			ObjectMeta: metav1.ObjectMeta{
-				Name: "ebs-claim",
-			},
-			StorageClassName: aws.String(storageClassName),
-			Resources:        v1.ResourceRequirements{Requests: v1.ResourceList{v1.ResourceStorage: resource.MustParse("5Gi")}},
+		It("should run a pod with a pre-bound persistent volume (non-existent storage class)", func() {
```

Comment:
Should we test the case with us safely handling the case where someone is using a dynamically created PVC with a non-existent storage class? I know we discussed this edge-case and it might be nice to have some handling for it in our tests. Also happy to punt on this one and save it for another PR but it's good to know that this is a possibility

Reply:
Do you mean with an ephemeral persistent volume (that makes sense to add). I don't think a dynamic PV with a non-existent storage class makes sense to test because it would fail to schedule. This should be covered in the functional testing.